src/server/middlewares/perIndexTierAccessMiddleware/index.js

```diff
@@ -0,0 +1,48 @@
+import config from '../../config';
+import { firstLetterUpperCase } from '../../utils/utils';
+import authMWResolver from '../authMiddleware/resolvers';
+import { tierAccessResolver, hideNumberResolver } from '../tierAccessMiddleware/resolvers';
+
+const queryTypeMapping = {};
+const aggsTypeMapping = {};
+const totalCountTypeMapping = {};
+
+config.esConfig.indices.forEach((item) => {
+  if (item.tier_access_level === 'private') {
+    queryTypeMapping[item.type] = authMWResolver;
+    aggsTypeMapping[item.type] = authMWResolver;
+  } else if (item.tier_access_level === 'regular') {
+    queryTypeMapping[item.type] = tierAccessResolver({
+      isRawDataQuery: true,
+      esType: item.type,
+      esIndex: item.index,
+    });
+    aggsTypeMapping[item.type] = tierAccessResolver({ esType: item.type, esIndex: item.index });
+    const aggregationName = `${firstLetterUpperCase(item.type)}Aggregation`;
+    totalCountTypeMapping[aggregationName] = {
+      _totalCount: hideNumberResolver(true),
+    };
+  } else if (item.tier_access_level === 'libre') {
+    // No additional resolvers necessary
+  } else {
+    throw new Error(`tier_access_level invalid for index ${item.type}. Either set all index-scoped tiered-access levels or a site-wide tiered-access level.`);
+  }
+}, {});
+
+const perIndexTierAccessMiddleware = {
+  Query: {
+    ...queryTypeMapping,
+  },
+  Aggregation: {
+    ...aggsTypeMapping,
+  },
+  ...totalCountTypeMapping,
+  RegularAccessHistogramForNumber: {
```

Comment:
I think these should be added into the schema on demand if there is at least one index is on `regular` level, if not, just omit them from thge schema